src/unitxt/operators.py

```diff
@@ -1822,17 +1823,24 @@ def process(self, stream: Stream, stream_name: Optional[str] = None) -> Generato
 
             if not self.calc_confidence_intervals:
                 metric.disable_confidence_interval_calculation()
-
+            multi_stream = MultiStream(
+                {
+                    "tmp": ListStream(
+                        instances_list=deepcopy(
+                            instances_upon_entrance_to_metrics_evaluations
+                        )
+                    )
+                }
+            )
             multi_stream = metric(multi_stream)
-            multi_stream = CopyFields(
-                field_to_field={f"{k}_orig": k for k in keys_to_restore}
-            )(multi_stream)
+            for evaluated_instance, freezed_instance in zip(
+                multi_stream["tmp"], instances_upon_entrance_to_metrics_evaluations
+            ):
+                freezed_instance["score"] = deepcopy(evaluated_instance["score"])
```

Comment:
Maybe add warning when score with specific name is being overridden?
https://github.com/IBM/unitxt/issues/1069

Reply:
UnitxtWarning("score x from metric x_metric override score y from metric y_metric, consider adding score prefix to one of the metrics.",  link_to_prefix_docs, )